experimental/dds/tree2/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,194 @@
+    Additionally, the resulting experiences need to be maintainable, including easy authoring of new features and reviewing of changes for possible collaboration issues.
+    For code authors, it should be easy to find the right way to do something, and know it will be robust.
+    For code reviews, it should be obvious in review, only looking at the part of the code change, if it might introduce a collaboration related issue (such as unexpected merge behavior, issues collaborating with previous or future versions of the app, inability to open old document etc.).
+    This needs to be true, even for a novice users of shared tree without experience writing collaborative experiences.
+
+    One way this is achieved is to ensure work and requirements specific to collaboration support will be easy to discover when relevant and easy learn about and handle when needed.
+    For example, if adding a new editing operation, it should be obvious how to ensure a the required application invariants hold—even across merges—, and what the implications are for deployment and cross version collaboration.
+
+-   Generalizes investments into improved user experiences across many different application.
+
+    As much as possible, shared tree should serve as a framework for sharing generalized collaborative logic, and a good default library of such logic.
+    For example merge resolution strategies, compatibility adapters, debug tools, common schema etc, can be generalized for easy use by all users of Shared Tree.
+    Users of shared tree can start by authoring their own specialized version of any of these (if needed), and then consider generalizing them as either independent libraries or contributing upstream.
+
+Another way to describe these goals is that Shared Tree needs to be able to be **adopted with confidence**.
+Once adopted, it must be **easily learnable** and **productive** for developers new to collaborative application while being **flexible** in what it supports and produce **robust** and **high quality** collaborative experiences.
+
+In addition to these overall goals, Shared Tree also needs to provide an MVP early in the development and internally be **maintainable** and **extensible**.
+This allows it to deliver value early, but also continue to be developed to reduce the development effort required of application authors to improve the experience it enables for end users.
+These internal requirements mostly follow from the above flexibility requirement, but are worth the extra emphasis.
+
+# Implications of these Goals
+
+Commonly desired collaboration features easy for applications to support, but when supporting them requires extra work, can be explicitly oped out of or differed to a future version.
+
+This include features like:
```

Comment:
strongly agree with this entire list; I also think somewhere under "implications" there needs to be something about special attention paid to ease-of-use/learnability/adoptability.